Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -224,13 +230,81 @@ public class ConfirmDetailsViewModel: ConfirmDetailsViewModelType, ConfirmDetail
       context.map { $0.confirmationLabelHidden }
     )
 
-    self.configurePledgeSummaryViewControllerWithData = Signal.combineLatest(
+    let pledgeTotalSummaryData = Signal.combineLatest(
       projectAndConfirmationLabelHidden,
       pledgeTotal
     )
     .map(unpack)
     .map { project, confirmationLabelHidden, total in (project, total, confirmationLabelHidden) }
     .map(pledgeSummaryViewData)
+
+    self.configurePledgeSummaryViewControllerWithData = pledgeTotalSummaryData
+
+    // MARK: Pledge + Rewards Summary Table
+
+    self.pledgeRewardsSummaryViewHidden = Signal.zip(context, baseReward)
+      .map { context, reward in
+        if context.isAny(of: .pledge, .updateReward) {
+          return reward.isNoReward
+        }
+
+        return context.expandableRewardViewHidden
+      }
+
+    let shippingSummaryData = Signal.combineLatest(
+      selectedShippingRule.skipNil().map(\.location.localizedName),
+      project.map(\.stats.omitUSCurrencyCode),
+      project.map { project in
+        projectCountry(forCurrency: project.stats.currency) ?? project.country
+      },
+      allRewardsShippingTotal
+    )
+    .map(PledgeShippingSummaryViewData.init)
+
+    let bonusOrPledgeUpdatedAmount = self.pledgeAmountDataSignal.map { $0.amount }
```

Comment:
Nit: naming here is slightly confusing. `pledgeAmountDataSignal` makes me assume this is a signal of numbers, not a signal of tuples